src/ex4/server/model/todolist-model.js

```diff
@@ -0,0 +1,95 @@
+import {readData, writeData} from './data-acess.js'
+
+export default class TodoListModel {
+    constructor() {
+        this.todoList = []
+    }
+
+    loadDataFromFile(){
+        
+        try {
+            const data = readData()
+            if(data === null){
+                this.todoList = []
+            }
+            else{
+                this.todoList = JSON.parse(data)
+            } 
+        } catch (err) {
+            console.error(err);
+        }
+    }
+
+    addData(enterValue){
+        this.todoList.push(enterValue)
+    }
+
+    removeData(index){
+        let copyArr = this.todoList
+        copyArr.splice(index,1)
+        this.todoList = copyArr
+    }
+
+    saveDataToFile(){ 
```

Comment:
is it really important for a user of this service to know how it's saved? if not, consider remove the toFile part as this is an implementation details and you want to be able to easily change it behind the scenes without refactoring large parts of the code